src/main/kotlin/no/nav/syfo/kafka/DialogmeldingProducer.kt

```diff
@@ -0,0 +1,78 @@
+package no.nav.syfo.brev.behandler
+
+import no.nav.helse.msgHead.XMLMsgHead
+import no.nav.syfo.kafka.DialogmeldingForKafka
+import no.nav.syfo.model.*
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.slf4j.LoggerFactory
+
+class DialogmeldingProducer(
+    private val kafkaProducerDialogmelding: KafkaProducer<String, DialogmeldingForKafka>,
+    private val enabled: Boolean,
+) {
+    fun sendDialogmelding(
+        receivedDialogmelding: ReceivedDialogmelding,
+        msgHead: XMLMsgHead,
+        journalpostResponse: JournalpostResponse,
+        vedleggListe: List<Vedlegg>?,
+    ) {
+        if (enabled) {
+            try {
+                val dialogmeldingForKafka = createDialogmeldingForKafka(
+                    receivedDialogmelding = receivedDialogmelding,
+                    msgHead = msgHead,
+                    journalpostResponse = journalpostResponse,
+                    vedleggListe = vedleggListe,
+                )
+                kafkaProducerDialogmelding.send(
+                    ProducerRecord(
+                        DIALOGMELDING_TOPIC,
+                        dialogmeldingForKafka.msgId,
+                        dialogmeldingForKafka,
+                    )
+                ).get()
+            } catch (e: Exception) {
+                log.error(
+                    "Exception was thrown when attempting to send dialogmelding with id {}: ${e.message}",
+                    receivedDialogmelding.msgId,
+                    e
+                )
+                throw e
+            }
+        }
+    }
+
+    fun createDialogmeldingForKafka(
+        receivedDialogmelding: ReceivedDialogmelding,
+        msgHead: XMLMsgHead,
+        journalpostResponse: JournalpostResponse,
+        vedleggListe: List<Vedlegg>?,
+    ): DialogmeldingForKafka {
+        val xmlMsgInfo = msgHead.msgInfo
```

Comment:
En annen ting rundt jeg kommenterte under:

Burde vi kanskje beholde struktur og navngiving fra EI_fellesformat når vi tar ting fra MQ -> Kafka i stedet for å lage vår egen struktur.

Vi kan f.eks lage et kafka-objekt som er:

KafkaObjekt{
EI_fellesformat-objekt {...} // alt fra XML minus vedlegg
..... info som vi legger til 1
..... info som vi legger til 2
..... info som vi legger til 3
}